base/rational.jl

```diff
@@ -620,7 +620,7 @@ function hash(x::Rational{<:BitInteger64}, h::UInt)
         end
     end
     h = hash_integer(pow, h)
-    h = hash_integer(num, h)
```

Comment:
What's going on here?

Reply:
this change to the generic method is motivated by helping `BigInt` specialization avoid a lot of pain. specifically, the fact that leading 0s are omitted from the hash for performance reasons. see this line here https://github.com/JuliaLang/julia/pull/58440/files#diff-4801bef8f66451cd027edfdc4f6eaa7ffcd055315010d338b63f3c77e00cdfaaR902 where we offset the `hash_bytes` pointer to the  `BigInt` limbs by the number of all-0 bytes.

when you commented, I was offsetting by whole all-0 words (64 bits), but on looking at it again I think bytes is sufficient